PyPDF2/_writer.py

```diff
@@ -411,6 +411,54 @@ def insertBlankPage(
         deprecate_with_replacement("insertBlankPage", "insert_blank_page")
         return self.insert_blank_page(width, height, index)
 
+    @property
+    def opening(self) -> Union[None, Destination, TextStringObject, ByteStringObject]:
+        """
+        property returning the opening Destination
+        returns None if no destination is set
+
+        (value stored in "/OpenAction" entry in the Pdf Catalog)
+        """
+        if "/OpenAction" not in self._root_object:
+            return None
+        oa = self._root_object["/OpenAction"]
+        if isinstance(oa, (str, bytes)):
+            return create_string_object(str(oa))
+        elif isinstance(oa, ArrayObject):
+            try:
+                page, typ = oa[0:2]  # type: ignore
+                array = oa[2:]
+                return Destination("OpenAction", page, typ, *array)  # type: ignore
+            except Exception:
+                raise Exception(f"Invalid Destination {oa}")
+        else:
+            return None
+
+    @opening.setter
+    def opening(self, dest: Union[None, str, Destination, PageObject]) -> None:
+        """
+        property setting the opening Destination or Page or NamedDest (`str`)
+        None:  removes the opening entry
+        :param destination:.
+
+        (value stored in "/OpenAction" entry in the Pdf Catalog)
+        """
+        if dest is None:
+            try:
+                del self._root_object["/OpenAction"]
+            except Exception:
```

Comment:
I'm not a fan of the bare `Exception` catch as failing to delete for something other than a `KeyError` strikes me as violating the user expectation of what this function does.

I'd say to only catch on `KeyError` and let all others bubble up:
```suggestion
            except KeyError:
```